src/api/lib/tasks/dev/requests.rake

```diff
@@ -199,5 +199,357 @@ namespace :dev do
+            user_id: find_user(params['by_user']).try(:id),
+            group_id: find_group(params['by_group']).try(:id),
+            project_id: find_project(params['by_project']).try(:id),
+            package_id: find_package(params['by_project'], params['by_package']).try(:id)
+          },
+          history_elements: prepare_history_elements_data(params['history'])
+        }
+      end
+
+      bs_request.reviews.destroy_all
+      review_params.each do |params|
+        review = bs_request.reviews.create!(params[:review])
+        create_history_elements(review, params[:history_elements])
+      end
+    end
+
+    def create_history_elements(review, params)
+      case review.state
+      when :accepted
+        klass = HistoryElement::ReviewAccepted
+      when :declined
+        klass = HistoryElement::ReviewDeclined
+      end
+      params.each do |param|
+        klass.create!(op_object_id: review.id, user_id: param[:user_id], comment: param[:comment])
+      end
+    end
+
+    def prepare_history_elements_data(params)
+      return [] if params.nil?
+
+      if params.is_a?(Array)
+        params.map do |param|
+          {
+            user_id: find_user(param['who']).try(:id),
+            description: param['description'],
+            comment: param['comment']
+          }
+        end
+      else
+        [{
+          user_id: find_user(params['who']).try(:id),
+          description: params['description'],
+          comment: params['comment']
+        }]
+      end
+    end
+
+    def create_bs_request(params)
+      bs_request = BsRequest.new(params[:bs_request])
+      bs_request.bs_request_actions.new(params[:bs_request_actions])
+
+      bs_request.save!
+      print_message("Successfully created request##{bs_request.number}")
+
+      bs_request
+    end
+
+    def clone_project(project_name:, users: [])
+      project = Project.find_by(name: project_name)
+      project = create(:project, name: project_name) if project.nil?
+      print_message("#{project.name} successfully created")
+
+      print_message("Cloning prjconfigs for #{project.name}")
+      request = make_api_request(url: "#{base_api_url}/source/#{project.name}/_config")
+      clone_prj_configs(config: request, project: project, comment: "Cloned from #{project.name}")
+
+      print_message("Cloning _meta for #{project.name}")
+      request = make_api_request(url: "#{base_api_url}/source/#{project.name}/_meta")
+      request_data = Xmlhash.parse(request)
+
+      create_users_and_groups_from_meta(request_data: request_data, users: users)
+      create_repositories_from_meta(request_data: request_data, target_project: project)
+      clone_meta(meta: request, comment: "Cloned _meta from #{project.name}", project: project)
+    end
+
+    def branch_package(source_project_name:, source_package_name:, target_project:)
+      package = Package.find_by_project_and_name(target_project, source_package_name)
+      package.destroy if package.present?
```

Comment:
You can use `force: 1` as `BranchPackage` parameter to do this.

Reply:
True. It works